internal/operator/cluster/upgrade.go

```diff
@@ -318,21 +319,15 @@ func deleteBeforeUpgrade(clientset kubeapi.Interface, clusterName, currentPrimar
 // deletion to complete before proceeding with the rest of the pgcluster upgrade.
 func deploymentWait(clientset kubernetes.Interface, namespace, deploymentName string, timeoutSecs, periodSecs time.Duration) string {
 	ctx := context.TODO()
-	timeout := time.After(timeoutSecs * time.Second)
-	tick := time.NewTicker(periodSecs * time.Second)
-	defer tick.Stop()
-
-	for {
-		select {
-		case <-timeout:
-			return fmt.Sprintf("Timed out waiting for deployment to be deleted: [%s]", deploymentName)
-		case <-tick.C:
-			_, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
-			if err != nil {
-				return fmt.Sprintf("Deployment %s has been deleted.", deploymentName)
-			}
-		}
+
+	if err := wait.Poll(periodSecs*time.Second, timeoutSecs*time.Second, func() (bool, error) {
```

Comment:
Hmm. `time.Duration * time.Duration`. It looks like the only two calls are a bit higher up in this file. Please fix this while you're here.